server/src/main/scala/actors/ShiftsActorBase.scala

```diff
@@ -24,11 +24,18 @@ case object ClearState
 
 case object GetFeedStatuses
 
-case class GetPortState(from: MillisSinceEpoch, to: MillisSinceEpoch)
+case class PointInTimeQuery(pointInTime: MillisSinceEpoch, query: PointInTimeAbleQuery)
 
-case class GetPortStateForTerminal(from: MillisSinceEpoch, to: MillisSinceEpoch, terminal: Terminal)
+trait PointInTimeAbleQuery {
```

Comment:
I think of the `point in time` as being the event time stamp that we are looking at a snapshot of - I don't see how it can have a start date and an end date.

Reply:
It can have a start and end time because that's the period of data you're looking at.

eg, I want to look at what 2020-01-01 (start) to 2020-01-05 (end) looked like on 2019-12-15 (point in time)

Reply:
ok, that makes sense, I'm not sure that this name conveys that though. You have the view date range, and the snapshot point in time.

Reply:
Maybe we should chat about it. To me it means a `point in time` query must have a `start/from` and `end/to` date for the period you're looking at for that point in time
Up until now we've not needed to specify the date range because we've been restricted to only being able to query the 'live' state which contained about 48 hours of data. Now we have complete daily history we are able to see what any date range looked like at any point in time

Reply:
This trait only contains the date range though, not the actual point in time that you are viewing from.

Reply:
Totally right! Sorry I was a bit thick on this one. I've renamed it `DateRangeLike` now. Phew!